contracts/0.8.25/upgrade/V3Template.sol

```diff
@@ -0,0 +1,464 @@
+// SPDX-FileCopyrightText: 2025 Lido <info@lido.fi>
+// SPDX-License-Identifier: GPL-3.0
+
+// See contracts/COMPILERS.md
+pragma solidity 0.8.25;
+
+import {IAccessControlEnumerable} from "@openzeppelin/contracts-v4.4/access/AccessControlEnumerable.sol";
+import {IBurner as IBurnerWithoutAccessControl} from "contracts/common/interfaces/IBurner.sol";
+import {ILido} from "contracts/0.8.25/interfaces/ILido.sol";
+import {IVersioned} from "contracts/common/interfaces/IVersioned.sol";
+import {IOssifiableProxy} from "./interfaces/IOssifiableProxy.sol";
+import {V3Addresses} from "./V3Addresses.sol";
+
+
+interface IPausableUntil {
+    function isPaused() external view returns (bool);
+    function getResumeSinceTimestamp() external view returns (uint256);
+    function PAUSE_INFINITELY() external view returns (uint256);
+}
+
+interface IPausableUntilWithRoles is IPausableUntil, IAccessControlEnumerable {
+    function RESUME_ROLE() external view returns (bytes32);
+    function PAUSE_ROLE() external view returns (bytes32);
+}
+
+interface IOperatorGrid is IAccessControlEnumerable {
+    function REGISTRY_ROLE() external view returns (bytes32);
+}
+
+
+interface IBaseOracle is IAccessControlEnumerable, IVersioned {
+    function getConsensusContract() external view returns (address);
+}
+
+interface IStakingRouter is IAccessControlEnumerable {
+    function REPORT_REWARDS_MINTED_ROLE() external view returns (bytes32);
+}
+
+interface IBurner is IBurnerWithoutAccessControl, IAccessControlEnumerable {
+    function REQUEST_BURN_SHARES_ROLE() external view returns (bytes32);
+    function REQUEST_BURN_MY_STETH_ROLE() external view returns (bytes32);
+    function isMigrationAllowed() external view returns (bool);
+}
+
+interface ILidoWithFinalizeUpgrade is ILido {
+    function finalizeUpgrade_v3(address _oldBurner, address[4] calldata _contractsWithBurnerAllowances) external;
+}
+
+interface IAccountingOracle is IBaseOracle {
```

Comment:
```suggestion
interface IBaseOracleWithFinalizeUpgrade is IBaseOracle {
```

Reply:
but `finalizeUpgrade_vX` belongs to a specific inheriting Oracle contract